hydrolib/hydromt_delft3dfm/dflowfm.py

```diff
@@ -704,12 +694,81 @@ def setup_1dboundary(
         # 4. set boundaries
         self.set_forcing(da_out, name=f"{da_out.name}_{branch_type}")
 
+
+    def setup_mesh2d(
+        self,
+        mesh2d_fn: str = None,
+        geom_fn: str = None,
+        bbox: list = None,
+        res: float = 100.0,
+    ):
+        """Creates an 2D unstructured mesh or prepares an existing 2D mesh according UGRID conventions.
+
+        An 2D unstructured mesh will be created as 2D rectangular grid from a geometry (geom_fn) or bbox. If an existing
+        2D mesh is given, then no new mesh will be generated
+
+        2D mesh contains ...
+
+        Note that:
+        (1) Refinement of the mesh is a seperate setup function, however a refined existing grid (mesh_fn) can already
+        be read.
+        (2) If no geometry, bbox or existing grid is specified for this setup function, then the self.region is used as
+         mesh extent to generate the unstructured mesh.
+        (3) Validation checks have been added to check if the mesh extent is within model region.
+        (4) Only existing meshed with only 2D grid can be read.
+        #FIXME: read existing 1D2D network file and extract 2D part.
+
+        Adds/Updates model layers:
+
+        * **1D2D links** geom: By any changes in 2D grid
+
+        Parameters
+        ----------
+        mesh2D_fn : str Path, optional
+            Name of data source for an existing unstructured 2D mesh
+        geom_fn : str Path, optional
+            Path to a polygon used to generate unstructured 2D mesh
+        bbox: list, optional
+            Describing the mesh extent of interest [xmin, ymin, xmax, ymax].
+        res: float, optional
+            Resolution used to generate 2D mesh. By default a value of 100 m is applied.
+
+        See Also
+        ----------
+
+        """
+        # Function moved to MeshModel in hydromt core
+        # Recreate region dict for core function
+        if mesh2d_fn is not None:
+            region = {"mesh": mesh2d_fn}
+        elif geom_fn is not None:
+            region = {"geom": str(geom_fn)}
+        elif bbox is not None:
+            region = {"bbox": bbox}
+        else:
+            raise ValueError(
```

Comment:
Do we also need to check the validity of the arguments? Such as that bbox should contain 4 values?

Reply:
Checked in the core super().setup_mesh